server/routerlicious/packages/services-ordering-rdkafka/src/rdkafkaProducer.ts

```diff
@@ -272,6 +275,15 @@ export class RdkafkaProducer extends RdkafkaBase implements IProducer {
 			};
 
 			const message = Buffer.from(JSON.stringify(boxcarMessage));
+			if (message.byteLength > this.producerOptions.maxMessageSize) {
+				const error = new NetworkError(
+					413,
+					// eslint-disable-next-line max-len
+					`Boxcar message size (${message.byteLength}) exceeded max message size (${this.producerOptions.maxMessageSize})`,
+				);
+				boxcar.deferred.reject(error);
```

Comment:
In line 305 we also call `handleError()` after `boxcar.deferred.reject(error)`. 
```ts
this.handleError(err, {
    restart: true,
    tenantId: boxcar.tenantId,
     documentId: boxcar.documentId,
});
```
I believe that eventually takes care of emitting the "error" even through the RdKafkaBase class: https://github.com/microsoft/FluidFramework/blob/b7452d6f94fad07998fe2c9318b6c65dadc73b34/server/routerlicious/packages/services-ordering-rdkafka/src/rdkafkaBase.ts#L127

Should we also add that here?

Reply:
HandleError restarts Kafka. Since we are catching the invalid message _before_ sending it to Kafka, it doesn't make sense to arbitrarily restart Kafka. Would emitting the error event also communicate that Kafka encountered an error and must restart? It seems like we want to avoid that entirely, since we are not actually experiencing an internal error here, and are simply rejecting an operation

Reply:
Makes total sense! I asked before thinking about it carefully, but I totally agree with both of you.